dataflow/src/main/java/org/checkerframework/dataflow/cfg/node/ObjectCreationNode.java

```diff
@@ -88,19 +163,37 @@ public boolean equals(@Nullable Object obj) {
         if (constructor == null && other.getConstructor() != null) {
             return false;
         }
+        if ((receiver != null && other.getReceiver() == null)
+                || (receiver == null && other.getReceiver() != null)
+                || (receiver != null
+                        && other.getReceiver() != null
+                        && !receiver.equals(other.getReceiver()))) {
+            return false;
+        }
 
         return getConstructor().equals(other.getConstructor())
                 && getArguments().equals(other.getArguments());
     }
 
     @Override
+    @SideEffectFree
     public int hashCode() {
-        return Objects.hash(constructor, arguments);
+        int result;
+        if (receiver != null) {
```

Comment:
Is this distinction necessary? What does `Objects.hash` do when an argument is `null`? The javadoc doesn't say, but I would assume it also just adds 0 to the hash. So the special case here seems unnecessary.